pkg/blockstorage/vmware/vmware.go

```diff
@@ -150,7 +178,29 @@ func (p *fcdProvider) SnapshotGet(ctx context.Context, id string) (*blockstorage
 }
 
 func (p *fcdProvider) SetTags(ctx context.Context, resource interface{}, tags map[string]string) error {
-	return errors.New("Not implemented")
+	switch r := resource.(type) {
+	case *blockstorage.Volume:
+		return p.setTagsVolume(ctx, r, tags)
+	case *blockstorage.Snapshot:
+		return nil
+	default:
+		return errors.New("Unsupported type for resource")
+	}
+}
+
+func (p *fcdProvider) setTagsVolume(ctx context.Context, volume *blockstorage.Volume, tags map[string]string) error {
+	if volume == nil {
+		return errors.New("Empty volume")
+	}
+	task, err := p.gom.UpdateMetadata(ctx, vimID(volume.ID), convertTagsToKeyValue(tags), nil)
+	if err != nil {
+		return errors.Wrap(err, "Failed to update metadata")
+	}
+	_, err = task.Wait(ctx, defaultWaitTime)
+	if err != nil {
+		return errors.Wrap(err, "Failed to wait on task")
```

Comment:
[minor] this method will return nil if err == nil. You don't need to check the error, you can just directly return `return errors.Wrap(err, "Failed to wait on task")`